docs/markdoc-component-documentation.md

```diff
@@ -0,0 +1,263 @@
+        "My sources say no.",
+        "Outlook not so good.",
+        "Outlook good.",
+        "Reply hazy, try again.",
+        "Yes – definitely.",
+        "You may rely on it.",
+        "Absolutely not!",
+        "Go for it!",
+        "No way, José!",
+        "Oh, hell no!",
+        "Hell yes!",
+        "Yes, if you believe hard enough.",
+        "No, but that shouldn't stop you.",
+        "Why would you even ask that?"
+    ]
+
+    while True:
+        question = input("Ask the Magic 8 Ball a question (type 'exit' to quit): ")
+        if question.lower() == 'exit':
+            print("Magic 8 Ball has left the chat.")
+            break
+        else:
+            print(random.choice(responses))
+
+if __name__ == "__main__":
+    magic_8_ball()
+
+```
+
+# Custom Markdoc Tags
+We've extended markdown with custom tags that let us make the docs richer and more interactive.
+
+## Admonitions
+
+Two types of admonitions are available. Warnings and notes.
+
+`{% warning %} This is a warning {% /warning %}`
+
+{% warning %} This is a warning {% /warning %}
+
+`{% note %} This is a note {% /note %}`
+
+{% note %} This is a note {% /note %}
+
+## Buttons
+Buttons are basically just links that are styled to look neat to the user and work for making links stand out.
+
+We have a few different styles of buttons that all do the same thing.
+
+### Default
+
+`{% button link="https://dog.ceo/" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" %} Click Me! {% /button %}
+
+### Primary
+
+`{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %}
+
+### Secondary
+
+`{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}`
+
+{% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %}
+
+### Blurple
+
+` {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} `
+
+{% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %}
+
+### Button Container
+The main use case I've seen for buttons in the docs is basically setting up multiple styling links in a row. Doing this requires putting buttons into a `ButtonContainer`
+
+{% buttonContainer %} {% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} {% /buttonContainer %}
+
+## Crosses and Checks 
+You can invoke a cross with this tag `{% cross /}%` and it looks like this {% cross /%}. You can invoke checks with this tag `{% check /}%` and it looks like this {% check /%}.
+
+Crosses and checks can also be used in lists:
+- {% check /%} Completed task
+- {% cross /%} Incomplete task
+- {% cross /%} Super Incomplete task 
+
+You can also put crosses and checks into headers 
+
+### This is a header with a check {% check /%}
```

Comment:
I'm on the fence if this is worth addressing - and it doesn't have to be part of this PR, since it feels like scope creep - but the current logic for heading IDs/anchors appends an extra hyphen when the heading includes a component.

For example, this is the ID for this heading:

```
#this-is-a-header-with-a-check-
```

This is avoidable if we set the heading ID manually, but IMO it's probably easier to trim these hyphens at some point.

Reply:
I'm sure there is something I can do to address this, and I would love to try. However, it is more economical to just custom-set the anchor links.

Reply:
Sounds good to me!